### DISCUSSION TITLE
How not to trigger a workflow by pushing a merge commit?

### DISCUSSION THREAD

**Discussion by antlu** [community/community#26228] [Actions]
<p>My workflow is triggered by both push and pull_request events.</p>
<p>Merging changes from a pull request fires a push event, which means the workflow runs again for a merge commit. This seems redundant.<br>
Is there a way not to run the workflow on merge?</p>

**Comment by BrightRan** [comment, marked as answer]
<p>When a push is from the merge of a PR, after the PR merged, on the target branch, it will automatically generate a head commit with the commit message looks like as " <strong>Merge pull request #&lt;PR_ID&gt; from &lt;source_branch&gt;</strong>".</p>
<p>You can use the property  <strong>github.event.head_commit</strong> of <a href="https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github context</strong></a> to access the  <strong>head_commit</strong>  object. And you also can use the property  <strong>github.event.head_commit.message</strong>  to view the message of the head commit.</p>
<p>So you can try using the if conditional like as below to skip executing all the jobs in your workflow.</p>
<pre><code>if: (github.event_name == 'push' &amp;&amp; contains(toJSON(github.event.head_commit.message), 'Merge pull request ') == false)
</code></pre>

**Reply by immjs** [reply]
Is this the only way? One can easily change this commit message...

**Comment by antlu** [comment]
<p>Thanks.</p>
<p>I did it without <strong>toJSON</strong> and with <strong>!</strong> instead of <strong>== false</strong>.</p>
